File: FOR/for.js
console.time('timer');
len = 100000001;
arr = new Array(len);
var i = 0;
while(i < len){
  arr[i] = i;
  i = (i+1)|0;
}
// i = 0;
// while(i < len){
//   arr[i] = i;
//   i = (i+1)|0;
// }
// i = 0;
// sum = 0;
// while(i < arr.length){
//   sum = (sum+arr[i])|0;
//   i=(i+1)|0;
// }
i = 0;
sum = 0;
while(i < len){
  sum += arr[i]|0; //ココを変えます
  i = (i+1)|0;
}
console.log(sum);
console.timeEnd('timer');

File: FOR/for.rb
=begin #若干遅くなるね
require "benchmark"
result = Benchmark.realtime do
  sum = 0; for i in 0..100000000
    sum += i
  end
end
puts "#{result}[s]"
=end

Start = Time.now
sum = 0; for i in 0..100000000
  sum += i
end
End = Time.now
p sum
puts "#{(End - Start) * 1000}[ms]"
